src/elements/OcDisclosureDrop.vue

```diff
@@ -0,0 +1,117 @@
+ *
+ * ## Accessibility
+ *
+ * ### When to use
+ * OcDisclosureDrop is useful for situations where you plan to hide/show content, links, form elements, media – in short: everything but a **pure** set of menu actions (for this, use OcActionDrop).
+ *
+ * ### DOM structure
+ * For keyboard and screen reader focus reasons the dropdown's container must follow the button directly in the DOM. Since other concepts of communicating the relation between button and its controlled containers are flawed or not supported by screen readers (`aria-controls`) the only way to show this relation is proximity in the DOM.
+ *
+ * ### Keyboard usage
+ * In comparison to OcActionDrop there is no need for special keyboard usage patterns or even programmatic focus management in disclosure widgets. When the dropdown's state is open, its focusable contents become part of the `tabindex`. If the state is closed, they are removed from it. The fact that button and controlled container are adjacent in the DOM makes sure that the next tab stop after the button after expanding is the first focusable element inside the container.
+ *
+ * */
+export default {
+  name: "oc-disclosure-drop",
+  status: "review",
+  release: "1.0.0",
+  data: () => {
+    return {
+      isOpen: false,
+    }
+  },
+  props: {
+    /**
+     * Supplies button variation to give additional meaning.
+     * `primary, secondary, danger`
+     */
+    buttonVariation: {
+      type: String,
+      default: "default",
+      validator: value => {
+        return value.match(/(default|primary|secondary|danger)/)
+      },
+    },
+  },
+  computed: {
+    $_ocButton_buttonClass() {
+      let classes = ["oc-button"]
+      if (this.buttonVariation) classes.push(`uk-button-${this.buttonVariation}`)
+
+      return classes
+    },
+  },
+  methods: {
+    toggleDropdown() {
+      this.isOpen = !this.isOpen
+    },
+    // Clicks everywhere outside the menu closes it
+    documentClick(e) {
+      let el = this.$refs.menu
+      let target = e.target
+      if (el !== target && !el.contains(target)) {
+        this.closeHandler()
+      }
+    },
+    closeHandler() {
+      this.isOpen = false
+    },
+  },
+  created() {
+    // Add a general event listener to catch outside clicks
+    document.addEventListener("click", this.documentClick)
+  },
+  destroyed: function() {
+    // Remove the general event listener to catch outside clicks
+    document.removeEventListener("keyup", this.documentClick)
+  },
+  mounted() {
+    // Add ESC key event listener
+    window.addEventListener("keyup", e => {
```

Comment:
I feel a little bit scared seeing this event which is not then removed.

Reply:
Updated: will get removed `beforeDestroy`